examples/chrono_binding.cpp

```diff
@@ -0,0 +1,169 @@
+    ss >> std::chrono::parse("%F"s, tt);
+    if(!ss.fail()) {
+        return {tt};
+    }
+    return std::nullopt;
+}
+
+/**
+ *  This is where magic happens. To tell sqlite_orm how to act
+ *  with SysDays we have to create a few service classes
+ *  specializations (traits) in sqlite_orm namespace.
+ */
+namespace sqlite_orm {
+
+    /**
+     *  First of all is a type_printer template class.
+     *  It is responsible for sqlite type string representation.
+     *  We want SysDays to be `TEXT` so let's just derive from
+     *  text_printer. Also there are other printers: real_printer and
+     *  integer_printer. We must use them if we want to map our type to `REAL` (double/float)
+     *  or `INTEGER` (int/long/short etc) respectively.
+     */
+    template<>
+    struct type_printer<std::chrono::sys_days> : public text_printer {};
+
+    /**
+     *  This is a binder class. It is used to bind c++ values to sqlite queries.
+     *  Here we have to create sysday string representation and bind it as string.
+     *  Any statement_binder specialization must have `int bind(sqlite3_stmt*, int, const T&)` function
+     *  which returns bind result. Also you can call any of `sqlite3_bind_*` functions directly.
+     *  More here https://www.sqlite.org/c3ref/bind_blob.html
+     */
+    template<>
+    struct statement_binder<std::chrono::sys_days> {
+
+        int bind(sqlite3_stmt* stmt, int index, const std::chrono::sys_days& value) const {
+            return statement_binder<std::string>().bind(stmt, index, sysDaysToString(value));
+        }
+    };
+
+    /**
+     *  field_printer is used in `dump` and `where` functions. Here we have to create
+     *  a string from mapped object.
+     */
+    template<>
+    struct field_printer<std::chrono::sys_days> {
+        std::string operator()(const std::chrono::sys_days& t) const {
+            return sysDaysToString(t);
+        }
+    };
+
+    /**
+     *  This is a reverse operation: here we have to specify a way to transform string received from
+     *  database to our sysdays object. Here we call `sysDaysFromString` and throw `std::runtime_error` if it returns null.
+     *  Every `row_extractor` specialization must have `extract(const char*)`, `extract(sqlite3_stmt *stmt, int columnIndex)`
+     *	and `extract(sqlite3_value* value)`
+     *  functions which return a mapped type value.
+     */
+    template<>
+    struct row_extractor<std::chrono::sys_days> {
+        std::chrono::sys_days extract(const char* row_value) const {
+            auto sd = sysDaysFromString(row_value);
+            if(sd) {
+                return sd.value();
+            } else {
+                throw std::runtime_error("incorrect date string (" + std::string(row_value) + ")");
+            }
+        }
+
+        std::chrono::sys_days extract(sqlite3_stmt* stmt, int columnIndex) const {
+            auto str = sqlite3_column_text(stmt, columnIndex);
+            return this->extract((const char*)str);
```

Comment:
Beware: `sqlite3_column_text` might return null.

Reply:
then, is this function a correct replacement?

```C++
std::chrono::sys_days extract(sqlite3_stmt* stmt, int columnIndex) const {
            auto str = sqlite3_column_text(stmt, columnIndex);
            if(str == nullptr)  str = "";
            return this->extract((const char*)str);
```

Reply:
no - forget this one!!

Reply:
right but please skip `if` operator without curly braces. It is a rule from the code style

Reply:
```C++
template<>
    struct row_extractor<std::chrono::sys_days> {
        std::chrono::sys_days extract(const char* row_value) const {
            if(row_value) {
                auto sd = sysDaysFromString(row_value);
                if(sd) {
                    return sd.value();
                } else {
                    throw std::runtime_error("incorrect date string (" + std::string(row_value) + ")");
                }
            } else {
                // ! row_value
                throw std::runtime_error("incorrect date string (nullptr)");
            }
        }
```

its done!

Reply:
> BTW extract throws exception is argument is null
yes what should it return?

Reply:
ok you hadn't seen this
we are ok now...